libutil/dr_frontend_unix.c

```diff
@@ -166,15 +168,18 @@ drfront_searchenv(const char *fname, const char *env_var, OUT char *full_path,
         NULL_TERMINATE_BUFFER(tmp);
         /* realpath checks for existence too. */
         if (realpath(tmp, realpath_buf) != NULL) {
-            status_check = drfront_access(realpath_buf, DRFRONT_EXIST, &access_ret);
+            status_check = drfront_access(realpath_buf, DRFRONT_EXEC, &access_ret);
             if (status_check != DRFRONT_SUCCESS) {
                 *ret = false;
                 return status_check;
             } else if (access_ret == true) {
-                *ret = true;
-                snprintf(full_path, full_path_size, "%s", realpath_buf);
-                full_path[full_path_size - 1] = '\0';
-                return DRFRONT_SUCCESS;
+                r = stat(realpath_buf, &st);
```

Comment:
> In this implementiation we are in the akward situation of calling stat() twice (first time in drfront_access() and then in drfront_searchenv()). An alternative approach would be to add a new variant for drfront_access_mode_t called DRFRONT_ISDIR. Though ISDIR isn't really an access mode.
> 
> So what would the preferable method be?

Hmm, adding `DRFRONT_ISDIR = 0x08` or something could break existing code that wants to include directories: it would have to be negated `DRFRONT_NOTDIR = 0x08` so the unset behavior is the legacy behavior: and then it would be less useful b/c it couldn't be used to find a directory, only to exclude them...so maybe `drfront_access_ex` is better which looks at `DRFRONT_ISDIR` while `drfront_access` ignores it.  Hmm.  I could also live with the extra `stat`.  How about just a `XXX` comment here about possibly avoiding the extra `stat` by either the `DRFRONT_NOTDIR` or the  `DRFRONT_ISDIR` plus `drfront_access_ex`.

Reply:
I added a comment.